build/NuGetAdditionalFiles/Microsoft.NETCore.Compilers.props

```diff
@@ -9,15 +9,17 @@
              AssemblyFile="$(MSBuildThisFileDirectory)..\tools\Microsoft.Build.Tasks.CodeAnalysis.dll" />
   <UsingTask TaskName="Microsoft.CodeAnalysis.BuildTasks.CopyRefAssembly"
              AssemblyFile="$(MSBuildThisFileDirectory)..\tools\Microsoft.Build.Tasks.CodeAnalysis.dll" />
+
   <PropertyGroup>
+    <!-- Don't use the compiler server by default in the NuGet package. -->
+    <UseSharedCompilation Condition="'$(UseSharedCompilation)' == ''">false</UseSharedCompilation>
     <CSharpCoreTargetsPath>$(MSBuildThisFileDirectory)..\tools\Microsoft.CSharp.Core.targets</CSharpCoreTargetsPath>
     <BasicCoreTargetsPath>$(MSBuildThisFileDirectory)..\tools\Microsoft.VisualBasic.Core.targets</BasicCoreTargetsPath>
   </PropertyGroup>
 
-  <!-- Set ToolPath/Exe to direct to the exes in this package -->
-  <PropertyGroup>
-    <!-- Compiler server is not currently supported on CoreCLR -->
-    <UseSharedCompilation>false</UseSharedCompilation>
+  <!-- Older versions of the CLI didn't set the DOTNET_HOST_PATH variable, which
+       means we have to use our shell wrappers and hope 'dotnet' is on the path. -->
+  <PropertyGroup Condition="'$(DOTNET_HOST_PATH)' == ''">
```

Comment:
Can we file an issue to track deleting this? Once we get past 2.1 we should no longer support this case.

Reply:
I think we'll have to support installing newer compilers on older runtimes for quite a while, no? Is there any reason not to support that, if this is the only complexity?

Reply:
I didn't think it was possible at all in older CLI. Could be wrong.